src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -463,15 +469,60 @@ private static bool IsAfterDot(ITextSnapshot snapshot, ITrackingSpan applicableT
             }
 
             return new FilteredCompletionModel(
-                GetHighlightedList(matchResults), index, filters,
+                GetHighlightedList(matchResults, filterText, highlightMatchingSpans, completionHelper), index, filters,
                 hardSelect ? UpdateSelectionHint.Selected : UpdateSelectionHint.SoftSelected,
                 centerSelection: true,
-                uniqueItem: moreThanOneMatchWithSamePriority ? null : bestMatchResult.GetValueOrDefault().VSCompletionItem);
+                uniqueItem: moreThanOneMatchWithSamePriority ? null : bestMatchResult.GetValueOrDefault().EditorCompletionItem);
         }
 
-        private static ImmutableArray<CompletionItemWithHighlight> GetHighlightedList(List<MatchResult> matchResults)
-            => matchResults.SelectAsArray(matchResult =>
-            new CompletionItemWithHighlight(matchResult.VSCompletionItem, matchResult.HighlightedSpans));
+        private static ImmutableArray<CompletionItemWithHighlight> GetHighlightedList(
+            List<MatchResult<VSCompletionItem>> matchResults,
+            string filterText,
+            bool highlightMatchingPortions,
+            CompletionHelper completionHelper)
+        {
+            return matchResults.SelectAsArray(matchResult =>
+            {
+                var highlightedSpans = GetHighlightedSpans(completionHelper, matchResult.EditorCompletionItem, filterText, highlightMatchingPortions, matchResult.RoslynCompletionItem, matchResult.PatternMatch);
```

Comment:
could you just pass the match along, and have GetHighlightedSpans extract everythign it needs from that?